### DISCUSSION TITLE
How to check the next.js with electron template application use SSR or not?

### DISCUSSION THREAD

**Discussion by liaodalin19903** [vercel/next.js#52762] [Help] [answered]
### Summary

I use create-next-app command created a nextjs application:

```
npx create-next-app --example with-electron-typescript electron-nextjs-demo
```

but how to check whether it use the SSR feature? 
I think it should use SSR by default.

but however I written a console.log() function to test in my page: 
```
import { Allotment } from "allotment";
import "allotment/dist/style.css";

// import dynamic from 'next/dynamic'

// const Layout = dynamic(() => import('../components/Layout'), {
//     ssr: false,
// })


const IndexPage = () => {

  console.log('test SSR or not')
  
  return (
    <>
      start 
      ...
      end
    </>
  )
}

export default IndexPage

```

then it log in the browser rather than IDE terminal:
![image](https://github.com/vercel/next.js/assets/12727582/948a25b5-af92-4128-b4b7-afd4030a6c27)

So, my think is wrong. the next.js with electron app use CSR, right? if is, how to set SSR?

**Comment by icyJoseph** [comment, marked as answer]
Next.js is a hybrid, specially in the Pages route, so it does both SSR and CSR, something fancy which is often called isomorphic rendering. 

Anyway, is it not logging at all in the server? I tested with `npx create-next-app --example with-electron-typescript with-electron-typescript-app`, and it worked fine, it does SSR and CSR. 

Perhaps you have `_app` implementing `getInitialProps`?
